src/model/solver.jl

```diff
@@ -0,0 +1,63 @@
+"Abstract supertype of all differential equation solvers."
+abstract type DiffSolver end
+
+"Empty solver for nonlinear discrete-time models."
+struct EmptySolver <: DiffSolver end
+get_solver_functions(::DataType, ::EmptySolver, f!, h!, _ ... ) = f!, h!
+
+
+struct RungeKutta <: DiffSolver
+    order::Int
+    supersample::Int
+    function RungeKutta(order::Int, supersample::Int)
+        if order ≠ 4
+            throw(ArgumentError("only 4th order Runge-Kutta is supported."))
+        end
+        if supersample < 1
+            throw(ArgumentError("supersample must be greater than 0"))
+        end
+        return new(order, supersample)
+    end
+end
+
+"""
+    RungeKutta(order::Int=4; supersample::Int=1)
+
+Create a Runge-Kutta solver with optional super-sampling.
+
+Only the fourth order Runge-Kutta is supported for now. The keyword argument `supersample`
+provides the number of internal steps (default to 1 step).
+"""
+function RungeKutta(order::Int=4; supersample::Int=1)
+    if order < 1
+        throw(ArgumentError("order must be greater than 0"))
+    end
+    if supersample < 1
+        throw(ArgumentError("supersample must be greater than 0"))
+    end
+    return RungeKutta(order, supersample)
+end
+
+function get_solver_functions(NT::DataType, ::RungeKutta, f!, h!, Ts, _ , nx, _ , _ )
+    f! = let fc! = f!, Ts=Ts, nx=nx
```

Comment:
I think this let block is unnecessary. I guess you have added it to avoid the closure bug, but that bug only appears if the captured variables have been modified in the function body before capture, which they have not been here.

Reply:
Yes exactly, I'm following the tips at https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-captured. I will do some benchmark and `@code_warntype` and remove the let block if it is not necessary.